MapboxNavigation/NavigationMapView.swift

```diff
@@ -171,45 +172,72 @@ open class NavigationMapView: MGLMapView {
         }
     }
     
-    func shape(describing route: Route) -> MGLShape? {
+    func shape(describing route: Route, legIndex: Int?) -> MGLShape? {
         guard let coordinates = route.coordinates else { return nil }
         
-        let congestionPerLeg = route.legs.flatMap {
-            $0.segmentCongestionLevels
-        }
+        var linesPerLeg: [[MGLPolylineFeature]] = []
+        var previousLegCongestionIndex = 0
         
-        let combinedCongestionLevel = Array(congestionPerLeg.joined()) // Flatten all leg nodes
-        let destination = coordinates.suffix(from: 1)
-        let segment = zip(coordinates, destination).map { [$0.0, $0.1] }
-        let congestionSegments = Array(zip(segment, combinedCongestionLevel))
-        
-        // Merge adjacent segments with the same congestion level
-        var mergedCongestionSegments = [CongestionSegment]()
-        for seg in congestionSegments {
-            let coordinates = seg.0
-            let congestionLevel = seg.1
-            if let last = mergedCongestionSegments.last, last.1 == congestionLevel {
-                mergedCongestionSegments[mergedCongestionSegments.count - 1].0 += coordinates
-            } else {
-                mergedCongestionSegments.append(seg)
+        for (index, leg) in route.legs.enumerated() {
+            let legCongestion = leg.segmentCongestionLevels!
+            let coordsForLeg = coordinates[previousLegCongestionIndex..<previousLegCongestionIndex + legCongestion.count + 1]
+            let combinedCongestionLevel = leg.segmentCongestionLevels!
+            let destination = coordinates.suffix(from: previousLegCongestionIndex + 1)
+            let segment = zip(coordsForLeg, destination).map { [$0.0, $0.1] }
+            let congestionSegments = Array(zip(segment, combinedCongestionLevel))
+            
+            previousLegCongestionIndex = legCongestion.count
+            
+            // Merge adjacent segments with the same congestion level
+            var mergedCongestionSegments = [CongestionSegment]()
+            for seg in congestionSegments {
+                let coordinates = seg.0
+                let congestionLevel = seg.1
+                if let last = mergedCongestionSegments.last, last.1 == congestionLevel {
+                    mergedCongestionSegments[mergedCongestionSegments.count - 1].0 += coordinates
+                } else {
+                    mergedCongestionSegments.append(seg)
+                }
+            }
+            
+            let lines = mergedCongestionSegments.map { (congestionSegment: CongestionSegment) -> MGLPolylineFeature in
+                let polyline = MGLPolylineFeature(coordinates: congestionSegment.0, count: UInt(congestionSegment.0.count))
+                polyline.attributes["congestion"] = String(describing: congestionSegment.1)
+                if let legIndex = legIndex {
+                    polyline.attributes[isOpaqueIdentifier] = index != legIndex
```

Comment:
🤔  I'm wondering if it would get confusing to the user if we added a transparency to both non active legs and completed routes.